Softeq.XToolkit.Common.Tests/Extensions/ListExtensionsTests/ListExtensionsTests.cs

```diff
@@ -46,5 +47,114 @@ public void AddItem_EmptyListFilledFluent_ReturnsListWithAddedItems()
+        {
+            IList<int> list = new List<int>();
+            IEnumerable<int> range = new List<int>() { 1, 2, 3, 4 };
+
+            list.AddRange(range);
+
+            Assert.Equal(range.Count(), list.Count);
+            var rangeArray = range.ToArray();
+            for (var i = 0; i < list.Count; i++)
+            {
+                Assert.Equal(rangeArray[i], list[i]);
+            }
+        }
+
+        [Fact]
+        public void AddRange_NotEmptyList_AddsItemsToList()
+        {
+            IList<int> list = new List<int> { 1, 2, 3 };
+            var initialCount = list.Count;
+            IEnumerable<int> range = new List<int>() { 1, 2, 3, 4 };
+
+            list.AddRange(range);
+
+            Assert.Equal(initialCount + range.Count(), list.Count);
+            var rangeArray = range.ToArray();
+            for (var i = initialCount; i < list.Count; i++)
+            {
+                Assert.Equal(rangeArray[i - initialCount], list[i]);
+            }
+        }
+
+        [Fact]
+        public void AddRange_NotEmptyListWithEmptyRange_DoesNotChangeList()
+        {
+            IList<int> list = new List<int> { 1, 2, 3 };
+            var initialCount = list.Count;
+            IEnumerable<int> range = Enumerable.Empty<int>();
+
+            list.AddRange(range);
+
+            Assert.Equal(initialCount, list.Count);
+        }
+
+        [Fact]
+        public void InsertRange_EmptyListWithZeroIndex_InsertsItemsIntoList()
+        {
+            const int Index = 0;
+            IList<int> list = new List<int>();
+            IEnumerable<int> range = new List<int>() { 1, 2, 3, 4 };
+
+            list.InsertRange(Index, range);
+
+            Assert.Equal(range.Count(), list.Count);
+            var rangeArray = range.ToArray();
+            for (var i = 0; i < list.Count; i++)
+            {
+                Assert.Equal(rangeArray[i], list[i]);
+            }
+        }
+
+        [Fact]
+        public void InsertRange_EmptyListWithIndexOutOfBounds_ThrowsException()
+        {
+            const int Index = 1;
+            IList<int> list = new List<int>();
+            IEnumerable<int> range = new List<int>() { 1, 2, 3, 4 };
+
+            Assert.Throws<ArgumentOutOfRangeException>(() =>
+            {
+                list.InsertRange(Index, range);
+            });
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(2)]
+        [InlineData(3)]
+        public void InsertRange_NotEmptyList_InsertsItemsIntoList(int index)
+        {
+            IList<int> list = new List<int> { 1, 2, 3 };
+            var initialCount = list.Count;
+            IEnumerable<int> range = new List<int>() { 1, 2, 3, 4 };
+
+            list.InsertRange(index, range);
+
+            Assert.Equal(initialCount + range.Count(), list.Count);
+            var rangeArray = range.ToArray();
+            for (var i = index; i < index + range.Count(); i++)
```

Comment:
in this case can be simple checked only result length and first inserted item by index